### DISCUSSION TITLE
SIgterm errors in tasks

### DISCUSSION THREAD

**Discussion by sannytee** [apache/airflow#21248] [Q&A]
Airflow version: 2.2.3
Deployment: Docker-based deployment
Executor: localExecutor

We use `KubernetesPodOperator` for most of our tasks and we frequently get this error
![Screenshot 2022-02-01 at 10 21 03](https://user-images.githubusercontent.com/28694656/151942436-cc818ad6-f5eb-4264-a6b5-d04fcb1e371d.png)

when I check the scheduler logs, I realise the error below is very frequently
<img width="847" alt="Screenshot 2022-02-01 at 10 20 18" src="https://user-images.githubusercontent.com/28694656/151942299-2b202d19-7bcc-41b1-aab3-bc32aaf90b69.png">

I have tried updating to airflow latest version several times and the error still persists.

**Comment by potiuk** [comment]
You should check what is really closing your tasks. I believe this might be a problem with your resources - most likely memory. It's likely (whatever your Docker deployment is) somethin that manages your docker deployment sees that there is no memory or CPU or other resources and kills the task. You need to find out what kills your task.

Also you might take a look at optimizing your DAGs - because the memory issue might for example be triggered by not-optimized DAG. Look here for the best practices around it - https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code - this is something that users don't realise might cause performance and resource issues - when you simply do "too much" in the top-level code of your DAG.

**Reply by sannytee** [reply]
Thanks @potiuk . increasing the following configuration in airflow seems to have solve the issue we have, we haven't  seen the error in a while:
* scheduler_health_check_threshold: from 30 secs to 240
* orphaned_tasks_check_interval: from 300 secs to 600
* job_heart_beat_sec: from 5 secs to 60
* scheduler_heartbeat_sec: from 5 secs to 60
